ionoscloud/resource_dbaas_mariadb_cluster.go

```
@@ -0,0 +1,238 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	response, _, err := client.CreateCluster(ctx, *cluster)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while creating a DBaaS MariaDB cluster: %w", err))
+	}
+	clusterID := *response.Id
+	d.SetId(clusterID)
+
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error occured while checking the status for MariaDB cluster with ID: %v, error: %w", clusterID, err))
+	}
+	if err := client.SetMariaDBClusterData(d, response); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func mariaDBClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).MariaDBClient
+	clusterID := d.Id()
+	_, apiResponse, err := client.DeleteCluster(ctx, d.Id())
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error while deleting MariaDB cluster with ID: %v, error: %w", clusterID, err))
+	}
+	err = utils.WaitForResourceToBeDeleted(ctx, d, client.IsClusterDeleted)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("deletion check failed for MariaDB cluster with ID: %v, error: %w", clusterID, err))
+	}
+
+	// wait 15 seconds after the deletion of the cluster, for the lan to be freed
+	time.Sleep(constant.SleepInterval * 3)
+
+	return nil
+}
+
+func mariaDBClusterImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(services.SdkBundle).MariaDBClient
+	clusterId := d.Id()
+	cluster, apiResponse, err := client.GetCluster(ctx, clusterId)
+
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil, fmt.Errorf("MariaDB cluster with ID: %v does not exist", clusterId)
+		}
+		return nil, fmt.Errorf("an error occured while trying to import MariaDB cluster with ID: %v", clusterId)
```

I think we need to actually throw the error

---

@cristiGuranIonos solved in a25d7c6c779850ffeb70f942e90a214806ee271d